soccer/gameplay/tactics/find_pass_point.py

```diff
@@ -0,0 +1,24 @@
+import robocup
+import constants
+from scipy.optimize import minimize
+
+## Function that optimizes cost function for passing
+# uses scipy minimize using the BFGS method
+def findPassPoint(heuristic) -> robocup.Point:
+    width = constants.Field.Width
+    length = constants.Field.Length
+    initial_guess = [0.00, 0.00]
+    result = minimize(heuristic, initial_guess, bounds=((-width/2.00, width/2.00), (0.00, length)), tol=1e-3)
+    if (not result.success):
+        raise OptimizationError()
```

Comment:
I'm okay with this raising an exception as long as the only thing that calls it directly is the pass tactic, but that's not how it's currently set up. See my above comment.

Otherwise, people are probably not going to stick the exception handling boilerplate everywhere they use this, because it's not a regular program case and people are lazy.

Reply:
Okay so the minimizer fails if it doesn't find the exact minima within the number of allotted iterations, but that's sort of the point of limiting iterations, so I'll get rid of the exception, but should I like print something that says that the minimizer failed?